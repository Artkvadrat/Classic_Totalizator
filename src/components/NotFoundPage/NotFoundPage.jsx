import React from 'react';
import { Link } from 'react-router-dom';

import styles from './NotFoundPage.module.css';
import NotFoundPageImage from './NotFoundPage.png';

const NotFoundPage = () => (
  <div>
    <h1 className={styles.text}>Page Not Found</h1>
    <div className={styles.image}>
      <img src={NotFoundPageImage} alt="Jean Claude Van Damme" />
    </div>
    <p className={styles.text}>
      Wrong Bet. <Link to="/">Go Home</Link>
    </p>
  </div>
);

export default NotFoundPage;
