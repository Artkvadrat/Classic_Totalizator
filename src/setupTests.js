import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

enzyme.configure({ adapter: new Adapter() });

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn()
};
window.localStorage = localStorageMock;
