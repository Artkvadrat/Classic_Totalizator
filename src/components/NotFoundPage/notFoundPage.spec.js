import React from 'react';
import { shallow } from 'enzyme';

import '../../setupTests';

import NotFoundPage from './NotFoundPage';

describe('NotFoundPage component', () => {
  const wrapper = shallow(<NotFoundPage />);

  it('should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct structure', () => {
    expect(wrapper.find('h1').prop('children')).toEqual('Page Not Found');
    expect(wrapper.find('img').length).toEqual(1);

    expect(wrapper.find('p').prop('children').length).toEqual(2);
  });

  it('should provide link to main page', () => {
    expect(wrapper.find('Link').prop('to')).toEqual('/');
  });
});
