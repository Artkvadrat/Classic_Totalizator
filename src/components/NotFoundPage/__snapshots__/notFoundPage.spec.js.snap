// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component should render without errors 1`] = `
<div>
  <h1
    className="text"
  >
    Page Not Found
  </h1>
  <div
    className="image"
  >
    <img
      alt="Jean Claude Van Damme"
      src={null}
    />
  </div>
  <p
    className="text"
  >
    Wrong Bet. 
    <Link
      to="/"
    >
      Go Home
    </Link>
  </p>
</div>
`;
